server/paymentProviders/paypal/index.js

```diff
@@ -125,23 +125,23 @@ export default {
             );
           }
 
+          const redirectUrl = new URL(paymentMethod.data.redirect);
+          redirectUrl.searchParams.set('paypalApprovalStatus', req.query.paypalApprovalStatus);
+
           if (req.query.paypalApprovalStatus !== 'success') {
             pm.destroy();
-            const redirect = `${paymentMethod.data.redirect}?status=error&service=paypal&error=User%20cancelled%20the%20request`;
-            return res.redirect(redirect);
+            redirectUrl.searchParams.set('paypalApprovalError', 'User cancelled the request');
+            return res.redirect(redirectUrl.toString());
           }
 
           return (
             getPreapprovalDetailsAndUpdatePaymentMethod(pm)
               .catch(e => {
                 debugPaypal('>>> paypal callback error:', e);
-                const redirect = `${
-                  paymentMethod.data.redirect
-                }?status=error&service=paypal&error=Error%20while%20contacting%20PayPal&errorMessage=${encodeURIComponent(
```

Comment:
All these statuses were not used in the frontend (so it should not be breaking), and the way we generated the URL was breaking if the redirect already contains filter parameters (which the new page does for filtering).